backend/mlarchive/archive/utils.py

```diff
@@ -448,3 +383,197 @@ def move_list(source, target):
+            is_known_email = bool(lookup_user(user.username))
+        except ValidationError:
+            is_known_email = False
+            logger.info(f'{user.username} is not a valid email. Not looking up.')
+        if not is_known_email:
+            count = count + 1
+            email = username_to_email(username=user.username)
+            if not email:
+                logger.warn(f'init_cheeck_users: no email found for {user.username}')
+                # logger.info(f'deleting user {user.username}')
+                # user.delete()
+                continue
+            logger.info(f'Found non-email user.username. Converting {user.username} to {email}')
+            logger.info(f'{user.username}=>{email}')
+            new_user, created = User.objects.get_or_create(username=email, defaults={'email': email})
+            emaillists = user.emaillist_set.all()
+            new_user.emaillist_set.add(*emaillists)
+            # assert user.last_login is None      # confirm never logged in
+            # logger.info(f'deleting user {user.username}')
+            # user.delete()
+    logger.info(f'{count} changed')
+
+
+def init_set_user_email():
+    '''Old Users created by get_subscribers didn't get email set.'''
+    for user in User.objects.filter(email=''):
+        user.email = user.username
+        user.save()
+
+
+def init_derived_mailmanmember():
+    mailman_lists = get_mailman_lists(private=True)
+    pks = [x.pk for x in mailman_lists]
+    non_mailman_lists = EmailList.objects.filter(private=True).exclude(pk__in=pks)
+    for elist in non_mailman_lists:
+        for member in elist.members.all():
+            MailmanMember.objects.create(email_list=elist, address=member.username)
+
+
+def lookup_user(address):
+    '''
+    This function takes an email address and looks in Datatracker for an associated
+    Datatracker account name.  Returns None if the email is not found or if there is no
+    associated User account.
+    '''
+    # check cache
+    username_map = cache.get('username_map')
+    if username_map and address in username_map:
+        return username_map[address]
+
+    apikey = settings.DATATRACKER_PERSON_ENDPOINT_API_KEY
+    url = settings.DATATRACKER_PERSON_ENDPOINT
+    data = {'apikey': apikey, '_expand': 'user', 'email': address}
+
+    try:
+        response = requests.post(url, data, timeout=settings.DEFAULT_REQUESTS_TIMEOUT)
+    except requests.exceptions.RequestException as error:
+        logger.error(str(error))
+        return None
+
+    if response.status_code != 200:
+        logger.error('Call to %s returned error %s' % (url, response.status_code))
+        return None
+
+    try:
+        output = response.json()
+        person_ids = list(output['person.person'])
+        if not person_ids:
+            logger.warning(f'lookup_user failed for {address}')
+            return None
+        user = output['person.person'][person_ids[0]]['user']
+        if user == 'None':
```

Comment:
Does the API really return the string "None" here? I"d have expected it to turn a "None" into a JSON null value and back to a `None` by the time you get it here.

Reply:
I don't see this addressed - and I think you get a `None`, not `'None'` here because the datatracker API returns `null` in the JSON:
```json
{
   "person.person": {
      "[ID of person without a user]": {
         "ascii": "[some ascii name]",
         "ascii_short": null,
         "biography": "",
         "name": "[some name]",
         "name_from_draft": "[some name]",
         "photo": "",
         "photo_thumb": "",
         "plain": "",
         "pronouns_freetext": null,
         "pronouns_selectable": "[]",
         "time": "2025-05-14T03:13:05.909Z",
         "user": null
      }
   }
}
```

Reply:
From external discussions, when `'_expand': 'user'` is included in the API call, the `None` expands into `"None"` so this is correct.